pymongo/client_session.py

```diff
@@ -439,31 +440,37 @@ def abort_transaction(self):
                 "Cannot call abortTransaction after calling commitTransaction")
 
         try:
-            self._finish_transaction_with_retry("abortTransaction")
+            self._finish_transaction_with_retry("abortTransaction", False)
         except (OperationFailure, ConnectionFailure):
             # The transactions spec says to ignore abortTransaction errors.
             pass
         finally:
             self._transaction.state = _TxnState.ABORTED
 
-    def _finish_transaction(self, command_name):
+    def _finish_transaction(self, command_name, retrying):
+        wc = self._transaction.opts.write_concern
+        if retrying and command_name == "commitTransaction":
+            wc_doc = wc.document
```

Comment:
Not sure about this, but do we want to be modifying `opts.write_concern.document` in-place? Should we be copying this dictionary before messing with it?